addon/metrics-adapters/google-analytics-four.js

```diff
@@ -0,0 +1,73 @@
+import { assert } from '@ember/debug';
+import { compact } from 'ember-metrics/-private/utils/object-transforms';
+import removeFromDOM from 'ember-metrics/-private/utils/remove-from-dom';
+import BaseAdapter from './base';
+
+export default class GoogleAnalyticsFour extends BaseAdapter {
+  toStringExtension() {
+    return 'GoogleAnalyticsFour';
+  }
+
+  install() {
+    const { id, options } = this.config;
+    const defaultOptions = {
+      send_page_view: false,
+    };
+    const compactedOptions = compact({
+      ...defaultOptions,
+      ...options,
+    });
+
+    assert(
+      `[ember-metrics] You must pass a valid \`id\` to the ${this.toString()} adapter`,
+      id
+    );
+
+    this._injectScript(id);
+
+    window.dataLayer = window.dataLayer || [];
+    this.gtag('js', new Date());
+    this.gtag('config', id, compactedOptions);
+  }
+
+  _injectScript(id) {
+    let script = document.createElement('script');
+    script.async = true;
+    script.src = `https://www.googletagmanager.com/gtag/js?id=${id}`;
+
+    document.head.appendChild(script);
+  }
+
+  gtag() {
+    window.dataLayer.push(arguments);
+
+    return arguments;
+  }
+
+  trackEvent(options = {}) {
+    const compactedOptions = compact(options);
+    const { event } = compactedOptions;
+
+    if (!event) {
+      return;
+    }
+
+    delete compactedOptions.event;
+
+    return this.gtag('event', event, compactedOptions);
+  }
+
+  trackPage(options = {}) {
```

Comment:
Thats a good question. I just saw, that you don't have to specify page_location and page_title. By default GA4 uses the window.location and document.title

https://support.google.com/analytics/answer/11403294?hl=en

Reply:
OK, so it seems like we ought to fallback to these defaults should `options.page/title` not be provided.

Skimming your linked doc it does seem like spreading the remaining options seems like a good idea though.

Reply:
After some additional research I added a new option autoTracking (enabled by default) which prevents the trackPage function to be called. That means by default we are using the GA4 auto tracking based on the browser history. However, if you want to track the page views manually, you need to disable autoTracking and configurate the [Enhanced measurement](https://support.google.com/analytics/answer/9216061) Options correctly to avoid double counting page views.

Reply:
So, after a long pause I'm working on the task again. To be even closer to GA4, I renamed the config autoTrackig to send_page_view.